scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -388,71 +390,106 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
+                )
+            }
+        }
+
+    /**
+     * Get the copyright findings from the given [result].
+     */
+    internal fun getCopyrightFindings(result: JsonNode): List<CopyrightFinding> =
+        result["files"].flatMap { file ->
+            val path = file["path"].textValue()
+
+            file["copyrights"]?.toList().orEmpty().flatMap { copyrights ->
+                val startLine = copyrights["start_line"].intValue()
+                val endLine = copyrights["end_line"].intValue()
+                // While ScanCode 2.9.2 was still using "statements", version 2.9.7 is using "value".
+                val statements = (copyrights["statements"] ?: listOf(copyrights["value"]))
+
+                statements.map { statement ->
+                    CopyrightFinding(
+                        statement = statement.textValue(),
+                        location = TextLocation(
+                            path = path,
+                            startLine = startLine,
+                            endLine = endLine
+                        )
+                    )
+                }
+            }
+        }
+
     /**
      * Get the license found in one of the commonly named license files, if any, or an empty string otherwise.
      */
-    internal fun getRootLicense(result: JsonNode): String {
+    internal fun getRootLicense(licenseFindings: List<LicenseFinding>): String =
         // TODO: This function should return a list of all licenses found in all license files instead of only a single
         //       license.
-
-        val rootLicenseFile = result["files"].singleOrNull { file ->
-            val path = file["path"].textValue()
+        licenseFindings.singleOrNull { finding ->
             try {
-                file["licenses"]?.singleOrNull() != null
-                        && LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(path)) }
+                LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(finding.location.path)) }
             } catch (e: InvalidPathException) {
                 false
             }
-        } ?: return ""
-
-        return getLicenseId(rootLicenseFile["licenses"].single())
-    }
+        }?.license ?: ""
 
     /**
      * Return the copyright statements in the vicinity, as specified by [toleranceLines], of [licenseStartLine] in the
      * file [path]. The default value of [toleranceLines] is set to 5 which seems to be a good balance between
      * associating findings separated by blank lines but not skipping complete license statements.
      */
     internal fun getClosestCopyrightStatements(
-        path: String,
-        copyrights: Collection<JsonNode>,
+        copyrights: List<CopyrightFinding>,
         licenseStartLine: Int,
         toleranceLines: Int = 5
     ): SortedSet<CopyrightFindings> {
+        require(
+            copyrights.map { it.location.path }.distinct().size <= 1,
+            { "Given copyright statements must all point to the same file." }
```

Comment:
This closure should be moved out of the parameter list, like `require() {}`.

Reply:
done.